### ISSUE TITLE
parol-ls ConnectionRefused OS error

### ISSUE THREAD

**Issue by nblei** [jsinger67/parol#160]
On parol-ls 0.14.0:

```
Starting parol language server
Error: Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }
```

Kernel 5.19.0-50-generic Ubuntu 22.04

**Comment by jsinger67** [comment]
@nblei ,
thank you for reporting this issue.
Did you use Visual Studio Code plus parol's VS Code extension?

**Comment by jsinger67** [comment]
I installed Ubuntu 22.04.2 LTS on a new virtual machine alongside with code and parol-ls 0.14.0.
In this configuration I couldn't reproduce your problem.

I think I need more information and maybe a little support from your side to be able to figure out what the cause may be.

**Comment by nblei** [comment]
Nah, I'm just dumb and wasn't using the --stdio option.  Please ignore.

**Comment by jsinger67** [comment]
So I assume you use neovim as your favorite editor, right? ,😉
Can I ask you how the overall experience with parol-ls is?
I have very little feedback about that.

**Comment by nblei** [comment]
Honestly I don't have much feedback.  I'm using it so I can make changes to Dalance's veryl project.  Its quite straightforward to make minor changes to the grammar.

**Comment by jsinger67** [comment]
Ok, fine.
Don't hesitate to contact me in case of any problems regarding parole or parole language server.
Good luck!

**Comment by nblei** [comment]
@jsinger67 --- Not an issue, but a feature that could be extremely useful:  auto-generation of tests based on the parol grammar file.  Dalance's veryl grammar is quite complex and the ad-hoc testcases are currently inadequate.  Being able to generate automatic tests (modulo some recursion limit) would be very useful.

**Comment by jsinger67** [comment]
@nblei ,
Good point. I will think about that.
A try in this direction had been implemented by me.
Former there was a language generator which was able to generate random strings from user's grammar.
Unfortunately I had to disable the language generator because of version conflicts (see issue #135).
Anyway, I think your proposal is worth thinking about. We have so much auto-generation, why not generate tests 😉
